Turner.Infrastructure.Mediator/Mediators/StaticDispatcher.cs

```diff
@@ -8,19 +8,25 @@ namespace Turner.Infrastructure.Mediator
 {
     public class StaticDispatcher : IStaticDispatcher
     {
-        private Dictionary<Type, Func<object, object>> _dispatchers;
-
+        private readonly Func<Type, object> _handlerFactory;
+        private readonly Dictionary<Type, Func<object, object>> _dispatchers;
+          
         public StaticDispatcher(Func<Type, object> handlerFactory, params Assembly[] assemblies)
         {
+            _handlerFactory = handlerFactory ?? throw new ArgumentNullException(nameof(handlerFactory));
+            _dispatchers = new Dictionary<Type, Func<object, object>>();
+
             Register(handlerFactory, assemblies);
         }
 
         public object Dispatch(object request) =>
-            _dispatchers[request.GetType()](request);
+            _dispatchers.ContainsKey(request.GetType()) 
+                ? _dispatchers[request.GetType()](request)
+                : Register(request.GetType())(request);
 
         private void Register(Func<Type, object> handlerFactory, IEnumerable<Assembly> assemblies)
         {
-            _dispatchers = new Dictionary<Type, Func<object, object>>();
+            _dispatchers.Clear();
 
             var classTypes = assemblies
```

Comment:
Could you please update: 
`.Where(type => type.IsClass && !type.IsAbstract && !type.IsInterface)`
to
`.Where(type => type.IsClass && !type.IsAbstract)`

If it's a class, I don't think it can be an interface. 
Was there a specific issue that required this?

Reply:
You're 100% correct!  That code went through a few iterations and removing that condition must've slipped my mind once the tests went green 😂 Good catch!